### ISSUE TITLE
`import/namespace` false positives with conditional exports

### ISSUE THREAD

**Issue by logaretm** [oxc-project/oxc#20704]
The import plugin's module resolver hardcodes `conditionNames` to `["module", "import"]` ([source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/service/runtime.rs#L307)). This causes false positives for packages that use [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) with conditions like `browser`, `node`, `types`, `default`, etc.

Many packages in the ecosystem define different entry points per condition. When oxlint resolves with only `["module", "import"]`, it may pick an entry point that doesn't export all symbols, leading to incorrect `"X" not found in imported namespace` errors.

**Repro**: https://github.com/logaretm/oxlint-repro-19939

`@sentry/nextjs` defines separate `browser` and `node` entry points via conditional exports. `captureRouterTransitionStart` is only exported from the `browser` entry. Since oxlint resolves with `["module", "import"]`, it picks the server/default entry and reports:

```
× eslint-plugin-import(namespace): "captureRouterTransitionStart" not found in imported namespace "@sentry/nextjs".
```

This is a false positive, the export exists under the `browser` condition.

Possibly a duplicate or related to: https://github.com/oxc-project/oxc/issues/13258 I created this for visibility. 

Usually with Eslint, users would typically switch to `eslint-import-resolver-typescript` or `(eslint-import-resolver-x)` to get those false positives eliminated.

**Comment by bastienlimbour** [comment]
I think I may be hitting a related case with `@rn-primitives/*@1.4.0`.

I added more details in #13258 because the issue looks closely related to oxlint resolving/analyzing the JS entrypoint instead of the `.d.ts` file:

https://github.com/oxc-project/oxc/issues/13258#issuecomment-4582968867

**Comment by samson501** [comment]
I'm having the same issue with chroma-js@3.2.0, which providers separate node.js and browser modules within the same package.
